server/models/video/video-channel-sync.ts

```diff
@@ -0,0 +1,177 @@
+
+  @AllowNull(false)
+  @Default(null)
+  @Is('VideoChannelExternalChannelUrl', value => throwIfNotValid(value, isUrlValid, 'externalChannelUrl', true))
+  @Column(DataType.STRING(CONSTRAINTS_FIELDS.VIDEO_CHANNEL_SYNCS.EXTERNAL_CHANNEL_URL.max))
+  externalChannelUrl: string
+
+  @CreatedAt
+  createdAt: Date
+
+  @UpdatedAt
+  updatedAt: Date
+
+  @ForeignKey(() => VideoChannelModel)
+  @Column
+  videoChannelId: number
+
+  @BelongsTo(() => VideoChannelModel, {
+    foreignKey: {
+      allowNull: false
+    },
+    onDelete: 'cascade'
+  })
+  VideoChannel: VideoChannelModel
+
+  @AllowNull(false)
+  @Default(VideoChannelSyncState.WAITING_FIRST_RUN)
+  @Is('VideoChannelSyncState', value => throwIfNotValid(value, isVideoChannelSyncStateValid, 'state'))
+  @Column
+  state: VideoChannelSyncState
+
+  @AllowNull(true)
+  @Column(DataType.DATE)
+  lastSyncAt?: Date
+
+  static listByAccountForAPI (options: Pick<AvailableForListOptions, 'accountId'> & {
+    start: number
+    count: number
+    sort: string
+  }) {
+    const query = {
+      offset: options.start,
+      limit: options.count,
+      order: getSort(options.sort),
+      include: [
+        {
+          model: VideoChannelModel,
+          required: true,
+          where: {
+            accountId: options.accountId
+          }
+        }
+      ]
+    }
+    return Promise.all([
+      VideoChannelSyncModel.count(query),
+      VideoChannelSyncModel.findAll(query)
+    ]).then(([ total, data ]) => ({ total, data }))
+  }
+
+  toFormattedJSON (this: MChannelSyncFormattable): VideoChannelSync {
+    return {
+      id: this.id,
+      state: {
+        id: this.state,
+        label: VIDEO_CHANNEL_SYNC_STATE[this.state]
+      },
+      externalChannelUrl: this.externalChannelUrl,
+      createdAt: this.createdAt.toISOString(),
+      channel: this.VideoChannel.toFormattedSummaryJSON(),
+      lastSyncAt: this.lastSyncAt?.toISOString()
+    }
+  }
+
+  static load (id: number) {
+    return this.findOne({
+      where: { id }
+    })
+  }
+
+  static loadWithAccount (id: number) {
+    return this.findOne({
+      where: { id },
+      include: [ {
+        model: VideoChannelModel,
+        required: true,
+        include: [ {
+          model: AccountModel,
+          required: true
+        } ]
+      } ]
+    })
+  }
+
+  static async listSyncs (): Promise<MChannelSyncChannel[]> {
+    const query = {
+      include: [
+        {
+          model: VideoChannelModel.unscoped(),
+          required: true,
+          include: [
+            {
+              model: AccountModel.unscoped(),
+              required: true,
+              include: [ {
+                attributes: [],
+                model: UserModel.unscoped(),
+                where: {
+                  videoQuota: {
+                    [Op.ne]: 0
+                  },
+                  videoQuotaDaily: {
+                    [Op.ne]: 0
+                  }
```

Comment:
These tests prevent that users without quota clutter the synchronization job.

I have this questions:
 - how many syncs we may allow by user? (20 syncs? More?)

Related: I implemented a check that the user asking for a channel import have some quota left: 1ad1ac2bd20a5b610ac35ff1a5322ae5d60affc2 (so we avoid abusive calls to yt-dlp / youtube-dl)

Reply:
> how many syncs we may allow by user? (20 syncs? More?)

You can add this as an admin config

Reply:
Alright. For the UI, does this sounds good to you?

![image](https://user-images.githubusercontent.com/371705/183041542-da7ce9e4-2edf-4f2b-92e0-fa8e673a1cd4.png)

Reply:
Finally I changed my mind: I made no UI, as I am not sure it is important enough. I can make one if you prefer.

Done in 37082c1d3e3040be8580fb5b53d92d32e3e9e626